llama_agi/llama_agi/execution_agent/base.py

```diff
@@ -0,0 +1,41 @@
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional, Union, TypedDict, NotRequired
+
+from langchain.agents.tools import Tool
+from langchain.llms import OpenAI, BaseLLM
+from langchain.chat_models.base import BaseChatModel
+from langchain.chat_models import ChatOpenAI
+
+
+class LlamaAgentPrompts(TypedDict):

```

Comment:
any reason this is a TypedDict and not a dataclass/pydantic model?

Reply:
Not sure, I'm still a noob at python typing LOL and it's what StackOverflow told me

I'll look into making it a dataclass instead, that does make more sense